mina-transaction/transaction-leaves.ts

```diff
@@ -70,3 +71,21 @@ const TransactionVersion = {
   ...provable(UInt32),
   empty: () => UInt32.from(protocolVersions.txnVersion),
 };
+
+type MayUseToken = {
+  parentsOwnToken: Bool;
+  inheritFromParent: Bool;
+};
+const MayUseToken = {
+  ...provable({
+    parentsOwnToken: Bool,
+    inheritFromParent: Bool,
+  }),
+  check: ({ parentsOwnToken, inheritFromParent }: MayUseToken) => {
+    parentsOwnToken
+      .and(inheritFromParent)
+      .assertFalse(
+        'MayUseToken: parentsOwnToken and inheritFromParent cannot both be true'
+      );
+  },
```

Comment:
the check method is not complete: it also has to check that `parentsOwnToken` and `inheritFromParent` are boolean (this is what the base check method would have done)